.github/workflows/ci-cd-workflow.yml

```diff
@@ -200,6 +198,8 @@ jobs:
           ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
           ECR_REPOSITORY: ${{ github.event.repository.name }}
         run: |
+          ./gradlew bootBuildImage -x bootJar --imageName=$ECR_REGISTRY/$ECR_REPOSITORY:${{ github.sha }}
```

Comment:
Why do we need to rebuild the image?
If we rebuild it we can't guarantee it will match the previous build, the whole point of retagging the build is so we know that we are reusing the previously tested image.

Reply:
Yes Andy, I understand that we would like to reuse the previously tested image. 
We encountered the image not found failure in the pipeline, then we have pushed some commits attempted to fix this issue, but still facing this issue. What we guess is it would do clean up in every step, so the previous image could not be found. 
May you suggest how we can resolve this please?

Reply:
May have to pull the image to tag it in that case.

Reply:
@DorisWongNHS Ignore my previous comment have a look at this https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-retag.html

Reply:
@Judge40 As suggested by Uzair, I have updated the code for retag. Not sure do we need to push separately though.
https://github.com/marketplace/actions/docker-image-retag
Please let me know if you have any comment. Thanks

Reply:
That action just does a `docker pull` then `docker tag`, I'd rather tmdo that ourselves if that's the approach we're going for. It adds a lot of download time if the image gets large though, the above technique to update the ECR manifest might be more efficient.